root/static/scripts/edit/externalLinks.js

```diff
@@ -476,20 +476,22 @@ function isEmpty(link) {
 
 function withOneEmptyLink(links, dontRemove) {
   let emptyCount = 0;
+  let canRemoveCount = 0;
   const canRemove = {};
 
   links.forEach(function (link, index) {
     if (isEmpty(link)) {
       ++emptyCount;
       if (index !== dontRemove) {
         canRemove[index] = true;
+        canRemoveCount++;
       }
     }
   });
 
   if (emptyCount === 0) {
     return links.concat(newLinkState({relationship: _.uniqueId('new-')}));
-  } else if (emptyCount > 1 && _.size(canRemove)) {
+  } else if (emptyCount > 1 && canRemoveCount > 0) {
```

Comment:
Adding a count variable didn't seem burdensome or affect readability at all, so I did that to avoid an extra allocation (since `Object.keys` has to build a new list of keys every time you call it).